packages/amplify-cli/src/commands/gen2-migration/codegen-generate/src/backend/synthesizer.ts

```diff
@@ -908,14 +908,14 @@ export class BackendSynthesizer {
     }
 
     // Identity pool overrides
-    if (renderArgs.auth?.guestLogin === false || (renderArgs.auth?.identityPoolName && !renderArgs?.auth?.referenceAuth)) {
-      const cfnIdentityPoolVariableStatement = this.createVariableStatement(
-        this.createVariableDeclaration('cfnIdentityPool', 'auth.resources.cfnResources.cfnIdentityPool'),
+    if (renderArgs.auth?.guestLogin === false && renderArgs.auth?.identityPoolName && !renderArgs?.auth?.referenceAuth) {
```

Comment:
Looking at the code, the constant is only used if `renderArgs.auth?.guestLogin === false`, which means that this entire code block should only execute if both conditions are met.